src/main/java/time_manager_UI/DisplayAvailabilityTimeBlockUI.java

```diff
@@ -0,0 +1,129 @@
+package time_manager_UI;
+import javax.swing.*;
+import javax.swing.table.DefaultTableCellRenderer;
+import javax.swing.table.DefaultTableModel;
+import javax.swing.table.TableColumnModel;
+import UI.CentralWindow;
+import UI.WindowManager;
+import use_cases.display_availability_timeblocks.*;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.List;
+
+
+/**
+ * A TimeBlockManagerUi that uses JFrame to show a window that illustrates the Availability of a User.
+ * Created:2022-11-26
+ * Last updated:2022-12-02
+ *
+ * @author Amir Bare
+ */
+public class DisplayAvailabilityTimeBlockUI extends CentralWindow implements AvailabilityUiInterface {
+    private final DisplayAvailabilityTimeBlockController displayController;
+    private JPanel mainPanel;
+    private JTable showTable;
+    private JButton backbutton;
+    private JButton setAvailabilityButton;
+    private final WindowManager programWindows;
+
+    /**
+     * Default constructor for the view availability window
+     */
+
+    public DisplayAvailabilityTimeBlockUI(WindowManager existingWindows,
+                                          DisplayAvailabilityTimeBlockController displayController) {
+        super();
+//         store reference to existing windows in program
+        this.programWindows = existingWindows;
+        this.programWindows.addWindow(WindowManager.AVAILABILITY_REFERENCE_KEY, this);
+        this.displayController = displayController;
+        this.uiSettings();
+        this.backbuttonListener();
+        this.setAvailabilityButtonListener();
+    }
+
+    /**
+     * Action Listener for the set availability button
+     */
+    private void setAvailabilityButtonListener() {
+        setAvailabilityButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                programWindows.closeWindow(WindowManager.AVAILABILITY_REFERENCE_KEY);
+                programWindows.openWindow(WindowManager.SET_AVAILABILITY_REFERENCE_KEY);
+            }
+        });
+    }
+
+    /**
+     * Action Listener for the back button
+     */
```

Comment:
Good job on your documentation! It's important that we maintain well-documented code for future collaborators to ensure our codebase is easy to understand and expand on, when necessary.